scripts/android.mk

```diff
@@ -56,4 +49,12 @@ $(TOOLCHAIN_FILE): $(NDK_HOME) scripts/android.mk
 	echo 'set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)' >> $@
 	@$(POST_RULE)
 
+$(ANDROID_NDK_HOME):
```

Comment:
Why move this block down here? The rule above depends (sometimes!) on this rule, so being able to read it from top to bottom is pretty nice, and it seems like an unrelated change?

Reply:
I can move it back up. It's from my searches about how to check if directory exists and as I didn't know if ordering matters I just did like the example.

Reply:
Ah, right. Yeah, I don't think the order of the rules should matter in general. In almost all cases, they're just looked up as-needed, so `./scripts/build-aarch64-linux-android release` will say we require `release`, then that sort of cascades..
`release: _build/$(TARGET)/tox4j/libtox4j-c$(DLLEXT)` ->
`_build/$(TARGET)/tox4j/libtox4j-c$(DLLEXT): $(PREFIX)/tox4j.stamp` ->
`$(PREFIX)/tox4j.stamp: $(BUILDDIR)/tox4j/Makefile` -> etc, until it reaches something that depends on ANDROID_NDK_HOME.

Reply:
Yeah it shouldn't matter,